tests/guestlog/guestlog.go

```diff
@@ -0,0 +1,267 @@
+package guestlog
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	expect "github.com/google/goexpect"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/clientcmd"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/testsuite"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+const cirrosStartupTimeout = 60
+const testString = "GuestConsoleTest3413254123535234523"
+
+var _ = Describe("[sig-compute]Guest console log", decorators.SigCompute, Serial, func() {
+
+	var (
+		virtClient kubecli.KubevirtClient
+		alpineVmi  *v1.VirtualMachineInstance
+		cirrosVmi  *v1.VirtualMachineInstance
+	)
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+
+		alpineVmi = libvmi.NewAlpine()
+		alpineVmi.Spec.Domain.Devices.AutoattachSerialConsole = pointer.P(true)
+		alpineVmi.Spec.Domain.Devices.LogSerialConsole = pointer.P(true)
+
+		cirrosVmi = libvmi.NewCirros()
+		cirrosVmi.Spec.Domain.Devices.AutoattachSerialConsole = pointer.P(true)
+		cirrosVmi.Spec.Domain.Devices.LogSerialConsole = pointer.P(true)
+	})
+
+	Describe("[level:component] Guest console log container", func() {
+		Context("set LogSerialConsole", func() {
+
+			var originalConfig v1.KubeVirtConfiguration
+			BeforeEach(func() {
+				originalConfig = *util.GetCurrentKv(virtClient).Spec.Configuration.DeepCopy()
+			})
+
+			AfterEach(func() {
+				tests.UpdateKubeVirtConfigValueAndWait(originalConfig)
```

Comment:
Is it necessary to restore the config "manually"? IIRC this is done by the test suite anyways?

Reply:
couldn't it affect other parallel tests due to out of order executions?

Reply:
done